packages/webapp/components/search/SearchV1Page.tsx

```diff
@@ -7,19 +7,53 @@ import {
 import { useChat } from '@dailydotdev/shared/src/hooks';
 import { SearchContainer } from '@dailydotdev/shared/src/components/search/SearchContainer';
 import { useRouter } from 'next/router';
+import { searchPageUrl } from '@dailydotdev/shared/src/graphql/search';
 import { getLayout as getMainLayout } from '../layouts/MainLayout';
 
 const SearchPage = (): ReactElement => {
   const router = useRouter();
-  const { data, queryKey, handleSubmit, isLoading } = useChat({
-    id: router?.query?.id as string,
-    query: router?.query?.q as string,
+  const query = router?.query?.q as string;
+  const sessionId = router?.query?.id as string;
+  const { data, isLoading, queryKey, handleSubmit } = useChat({
+    id: sessionId,
   });
-  const content = data?.chunks?.[0]?.response || '';
+  const searchId = data?.id;
+  const chunk = data?.chunks?.[0];
+  const content = chunk?.response || '';
+
+  useEffect(() => {
+    if (!searchId) {
+      return;
+    }
+
+    const newRoute = `${searchPageUrl}?id=${searchId}`;
+
+    router.replace(newRoute, undefined, {
+      shallow: true,
+    });
+    // router reference is not stable https://github.com/vercel/next.js/issues/18127
```

Comment:
Oh, cool to know. Should we keep this on moving forward for other dependencies with `router` object?

Reply:
If we are using just the methods then yes, if we use some value like `router.query.something`, then we should include. It is fixed in Next 13 so hopefully down the line we upgrade and can use the new hook 🤞